File: php/src/Model/PreviewedWorkerWeights.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Current vs learned routing weights for one worker.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class PreviewedWorkerWeights
{
    public function __construct(
        public readonly string $workerId,
        /** @var array<string, float> */
        public readonly array $current,
        /** @var array<string, float> */
        public readonly array $learned,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            current: $data['current'] ?? [],
            learned: $data['learned'] ?? [],
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/PreviewedWorkerWeights.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * Current vs learned routing weights for one worker.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class PreviewedWorkerWeights {
    private String workerId;
    private Map<String, Double> current;
    private Map<String, Double> learned;

    public String getWorkerId() { return workerId; }
    public Map<String, Double> getCurrent() { return current; }
    public Map<String, Double> getLearned() { return learned; }
}
